### ISSUE TITLE
Unable to launch the repl

### ISSUE THREAD

**Issue by abhi18av** [Dierk/frepl-gui#9]
Hi Dierk)

I've been trying to get Frege repl up and running but I get the following error 

```
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.NoClassDefFoundError: javafx/scene/control/TextInputControl
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
	at java.lang.Class.getMethod0(Class.java:3018)
	at java.lang.Class.getMethod(Class.java:1784)
	at sun.launcher.LauncherHelper.validateMainClass(LauncherHelper.java:544)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:526)
Caused by: java.lang.ClassNotFoundException: javafx.scene.control.TextInputControl
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
```

I've pretty much tried all combinations of the `./frepl` command with `java` but still, no progress.
```
309  java frepl
  314  ./bin/frepl
  316  java ./bin/frepl
  317  javac ./bin/frepl
  318  sudo ./bin/frepl

```

Am I doing something silly here?

**Comment by Dierk** [comment]
can you please run `java -version` ?

**Comment by abhi18av** [comment]
Yeah, sure.

``` bash
 java -version

openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-8u91-b14-3ubuntu1~16.04.1-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)
```

**Comment by Dierk** [comment]
ok, that looks good and TextInputControl should be on the classpath.
How did you install? Following the download/install instructions from https://github.com/Dierk/frepl-gui/ or by building yourself?

**Comment by abhi18av** [comment]
No, I just went with th download. You feel building might solve the problem, or perhaps it's a path issue?

**Comment by Dierk** [comment]
I just checked and the download is good. Manual building is not required.
What OS are you on?

**Comment by Dierk** [comment]
oh - just run via `bin/frepl` not `java bin/frepl` :-)

**Comment by abhi18av** [comment]
Well, I tried `bin/repl` and  `/bin/repl` from `~/Downloads/frepl-0.6.0/frepl`, still the same error.

The download is self-sufficient right? Or do I need to have the `frege-master` somewhere around as well?

**Comment by Dierk** [comment]
the download is fully self-contained (except the java runtime). All required jars are in the lib folder.

In your download folder (just above /bin and /lib) try the following command: `java -cp ./lib/frepl-0.6.0.jar:./lib/jfxrt.jar:./lib/fregefx-0.2.0-SNAPSHOT.jar:./lib/frege-repl-core-1.2.1-SNAPSHOT.jar:./lib/frege-3.22.524-gcc99d7e.jar:./lib/frege-interpreter-core-1.2.1-SNAPSHOT.jar:./lib/frege-repl-nativedeps-1.2.1-SNAPSHOT.jar:./lib/jline-2.13.jar:./lib/frege-interpreter-java-support-1.2.1-SNAPSHOT.jar:./lib/jansi-1.11.jar:./lib/ecj-4.2.2.jar org.frege.Application`

**Comment by abhi18av** [comment]
Sadly, much the same result.

``` bash
➜  frepl pwd
/home/abhi18av/Downloads/frepl-0.6.0/frepl
➜  frepl java -cp ./lib/frepl-0.6.0.jar:./lib/jfxrt.jar:./lib/fregefx-0.2.0-SNAPSHOT.jar:./lib/frege-repl-core-1.2.1-SNAPSHOT.jar:./lib/frege-3.22.524-gcc99d7e.jar:./lib/frege-interpreter-core-1.2.1-SNAPSHOT.jar:./lib/frege-repl-nativedeps-1.2.1-SNAPSHOT.jar:./lib/jline-2.13.jar:./lib/frege-interpreter-java-support-1.2.1-SNAPSHOT.jar:./lib/jansi-1.11.jar:./lib/ecj-4.2.2.jar org.frege.Application
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.NoClassDefFoundError: javafx/scene/control/TextInputControl
    at java.lang.Class.getDeclaredMethods0(Native Method)
    at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
    at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
    at java.lang.Class.getMethod0(Class.java:3018)
    at java.lang.Class.getMethod(Class.java:1784)
    at sun.launcher.LauncherHelper.validateMainClass(LauncherHelper.java:544)
    at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:526)
Caused by: java.lang.ClassNotFoundException: javafx.scene.control.TextInputControl
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 7 more


```

**Comment by Dierk** [comment]
Hm, weird. Did you run any other JavaFX application before on your machine? (I currently think your JRE may be defunct).

**Comment by abhi18av** [comment]
Oh, that might be possible. Will check it on the other machine then, btw using oracle JDK or OpenJDK doesn't make much of a difference right?

And are there any examples of the emitted `*.java` after compilation?

**Comment by Dierk** [comment]
The JDK makes no difference (7+). 
When the repl runs (or when using try.frege-lang.org), just type `:java` and you see the java code that your frege code emits.

**Comment by abhi18av** [comment]
Yup, it's exactly what you suspected. Something's wrong with the OpenJDK I installed on the Ubuntu machine. It works fine on mac.

Thanks @Dierk , I appreciate your time 😄

**Comment by Dierk** [comment]
Cool. Have fun!

**Comment by abhi18av** [comment]
@Dierk, I have another query but this one's regarding the `groovy` language classpath. I recently realized that you are one of the authors of the book I'm learning from ( ! ) so I thought it's best to ask you :)

I wish to use the `~/.m2/` in `groovy` on a global basis for ex. I need to use the `Apache Commons Math` from `groovy` but I'm unable to import it. Then I realized that `leiningen` took care of the classpath for my prior work with `jvm` so I narrowed it down to `.groovy/conf/`. This is where I'm stuck.

How to add the latest version of a lib in `.m2/` to my global groovy classpath - Is adding these libs manually line by line the way to do it? There ought to be a better one
